packages/core/streams/google-generative-ai-stream.ts

```diff
@@ -0,0 +1,60 @@
+import {
+  createCallbacksTransformer,
+  readableFromAsyncIterable,
+  type AIStreamCallbacksAndOptions,
+} from './ai-stream';
+import { createStreamDataTransformer } from './stream-data';
+
+interface GenerateContentResponse {
+  candidates?: GenerateContentCandidate[];
+}
+
+interface GenerateContentCandidate {
+  index: number;
+  content: Content;
+}
+
+interface Content {
+  role: string;
+  parts: Part[];
+}
+
+type Part = TextPart | InlineDataPart;
+
+interface InlineDataPart {
+  text?: never;
+}
+
+interface TextPart {
+  text: string;
+  inlineData?: never;
+}
+
+async function* streamable(response: {
+  stream: AsyncIterable<GenerateContentResponse>;
+}) {
+  for await (const chunk of response.stream) {
+    const parts = chunk.candidates?.[0].content.parts;
```

Comment:
I have gotten some occasional errors from this line: 
`⨯ Error: failed to pipe response`
`[cause]: TypeError: Cannot read properties of undefined (reading 'parts')`

Adding optional chaining here fixes it for me. 

```ts
 const parts = chunk.candidates?.[0]?.content?.parts;
```

Reply:
https://github.com/vercel/ai/pull/872